.github/workflows/build-and-deploy.yml

```diff
@@ -37,7 +37,7 @@ env:
 
 jobs:
   build:
-    runs-on: windows-latest
+    runs-on: ${{ github.event.inputs.architecture == 'aarch64' && fromJSON('["Windows", "ARM64"]') || 'windows-latest' }}
```

Comment:
I couldn't use `env.ARCHITECTURE` here so had to resort to `github.event.inputs.architecture`.

Reply:
Right, the environment requires the runner to already run, and that requires `runs-on` to be specified ;-)